cmake/target_toolchain.cmake

```diff
@@ -42,20 +42,20 @@ endif()
 unset(CMAKE_C_COMPILER)
 unset(CMAKE_C_COMPILER CACHE)
 
-# Configure the toolchain based on what toolchain technology is used
-# (gcc, host-gcc etc.)
+# A toolchain consist of a compiler and a linker.
+# In Zephyr, toolchains require a port under cmake/toolchain/.
+# Each toolchain port must set COMPILER and LINKER.
+# E.g. toolchain/llvm may pick {clang, ld} or {clang, lld}.
 include(${ZEPHYR_BASE}/cmake/compiler/${COMPILER}/target.cmake OPTIONAL)
+include(${ZEPHYR_BASE}/cmake/linker/${LINKER}/target.cmake OPTIONAL)
 
 # Uniquely identify the toolchain wrt. it's capabilities.
 #
 # What we are looking for, is a signature definition that is defined
 # like this:
-
-# Toolchains with the same signature will always support the same set
-# of flags.
-
+#  * Toolchains with the same signature will always support the same set
+#    of flags.
```

Comment:
Good opportunity to add the linker in the signature/checksum @SebastianBoe ?

Reply:
AFAICT this is correct, the linker should be a part of the toolchain signature.